src/vs/base/browser/ui/menu/menu.ts

```diff
@@ -200,26 +193,14 @@ export class Menu extends ActionBar {
 		const scrollElement = this.scrollableElement.getDomNode();
 		scrollElement.style.position = null;
 
-		menuElement.style.maxHeight = `${Math.max(10, window.innerHeight - container.getBoundingClientRect().top - 30)}px`;
-
-		this.menuDisposables.add(this.scrollableElement.onScroll(() => {
-			this._onScroll.fire();
-		}, this));
+		this.viewItems.filter(item => !(item instanceof MenuSeparatorActionViewItem)).forEach((item: BaseMenuActionViewItem, index: number, array: any[]) => {
+			item.updatePositionInSet(index + 1, array.length);
+		});
 
-		this._register(addDisposableListener(this.menuElement, EventType.SCROLL, (e: ScrollEvent) => {
-			if (this.scrollTopHold !== undefined) {
-				this.menuElement.scrollTop = this.scrollTopHold;
-				this.scrollTopHold = undefined;
-			}
-			this.scrollableElement.scanDomNode();
-		}));
+		this.push(actions, { icon: true, label: true, isMenu: true });
```

Comment:
`this.scrollableElement` must be created before calling this because now it's being used directly in `get onScroll()`